include/irr/core/math/aabb2dSIMDIntBase.h

```diff
@@ -0,0 +1,119 @@
+#ifndef __IRR_AABB_SIMD_INT_BASE_H_INCLUDED__
+#define __IRR_AABB_SIMD_INT_BASE_H_INCLUDED__
+
+#include "irr/core/math/irrMath.h"
+#include "irr/core/memory/memory.h"
+#include "irr/core/alloc/AlignedBase.h"
+#include "C:\IrrlichtBAW\IrrlichtBAW\include\vectorSIMD.h"
+
+#include <iostream>
+
+namespace irr { namespace core {
+
+//will modify this class so aabbSIMDf will be able do derive from it too
+template<template<typename> typename CRTP, typename VECTOR_TYPE>
+class aabbSIMDBase
+{
+	static_assert(	
+		std::is_same<VECTOR_TYPE, core::vector4di32_SIMD>::value ||
+		std::is_same<VECTOR_TYPE, core::vector4du32_SIMD>::value, 
+		"assertion failed: cannot use this type");
+
+public:
+	inline void addPoint(const VECTOR_TYPE& other)
+	{
+		static_cast<CRTP<VECTOR_TYPE>*>(this)->addBox();
+	}
+
+	inline VECTOR_TYPE getExtent() const
+	{
+		return getMaxEdge() - getMinEdge();
+	}
+
+	inline VECTOR_TYPE getMinEdge() const
+	{
+		VECTOR_TYPE minEdge = box;
+		minEdge.makeSafe2D();
+		return minEdge;
+	}
+
+	inline VECTOR_TYPE getMaxEdge() const
+	{
+		VECTOR_TYPE maxEdge = box.zwzw();
+		maxEdge.makeSafe2D();
+		return maxEdge;
+	}
+
+	inline CRTP<VECTOR_TYPE> reset(const VECTOR_TYPE& point) const
+	{
+		box = point.xyxy();
```

Comment:
reset can be done as an move assignment (`operator=(&&)`) of a new box created from a point